src/services/date-formatter/date-formatter.svc.coffee

```diff
@@ -0,0 +1,41 @@
+angular
+  .module('impac.services.date-formatter', [])
+  .service 'ImpacDateFormatter', ($log, $locale, ImpacTheming, $translate) ->
+
+    @getFormatForEntity = (entity = '') ->
+
+      settings = ImpacTheming.get()
+      if settings.dateFormatterSettings && settings.dateFormatterSettings[entity]
+        format = settings.dateFormatterSettings[entity]
+      else
+        format = 'L'
+      return format
+
+    @formatDateString = (obj, widgetEndpoint = '') ->
```

Comment:
This method is very obscure... :/ I think you are trying to perform a generic conversion of all the date members of any object (tell me if I'm wrong).
However I don't think this is the way to go: what if your object is something like:
```coffeescript
object:
  content:
    values:
      dates: ['2015-05-01', '2015-06-01']
```
? I don't think the dates would be converted in this case.

Instead, I would suggest to format the date **when it is displayed**, directly in the view.